stable-check/src/main.rs

```diff
@@ -26,15 +26,15 @@ fn check_directory(dir: &Path) -> Result<(), Box<dyn Error>> {
         let path = entry.path();
 
         if path.is_dir() {
-            return check_directory(&path);
-        }
-
-        let mut file = File::open(&path)?;
-        let mut contents = String::new();
-        file.read_to_string(&mut contents)?;
+            check_directory(&path)?;
+        } else {
+            let mut file = File::open(&path)?;
+            let mut contents = String::new();
+            file.read_to_string(&mut contents)?;
```

Comment:
You could use `fs::read_to_string`.

Reply:
Yea, I noticed that, too. I rewrote it in #886.